ocaml/xapi/xapi_host.ml

```diff
@@ -1346,9 +1346,23 @@ let apply_edition_internal  ~__context ~host ~edition ~additional =
     | Missing_connection_details -> raise Api_errors.(Server_error(missing_connection_details, []))
     | License_checkout_error s -> raise Api_errors.(Server_error(license_checkout_error, [s]))
   in
-  Db.Host.set_edition ~__context ~self:host ~value:edition';
-  let features = Features.of_assoc_list features' in
-  copy_license_to_db ~__context ~host ~features ~additional
+
+  let update_feature fname fenabled host =
+    Db.Feature.get_all_records ~__context |>
```

Comment:
It is quite inefficient to get all feature records in each iteration. Please factor this out off the helper function. Even better would be to use `get_all_records_where` to get all current records for a host.